src/main/java/nz/ac/auckland/scriptella/driver/http/HTTPConnection.java

```diff
@@ -1,288 +0,0 @@
-     */
-    public HTTPConnection(String host, String type, String format, int timeOut) {
-        this.host = host;
-        this.type = type;
-        this.format = format;
-        this.timeOut = timeOut;
-    }
-
-    /**
-     * Constructor used by Scriptella to initialise the HTTPConnection object.
-     *
-     * @param connectionParameters The parameters specified in the connection tag in the ETL file.
-     */
-    public HTTPConnection(ConnectionParameters connectionParameters) {
-
-        super(Driver.DIALECT, connectionParameters);
-
-        // Convert connection parameters to a map so that fields can be set.
-        Properties properties = CollectionUtils.asProperties(connectionParameters.getProperties());
-        properties.putAll(connectionParameters.getUrlQueryMap());
-
-        host = connectionParameters.getUrl();
-        logger.debug("Host: {}", host);
-
-        type = properties.getProperty("type", "GET");
-        logger.debug("Type: {}", type);
-
-        format = properties.getProperty("format", "String");
-        logger.debug("Format: {}", format);
-
-        timeOut = Integer.parseInt(properties.getOrDefault("timeout", DEFAULT_TIMEOUT).toString());
-        logger.debug("Timeout: {}", timeOut);
-
-    }
-
-    /**
-     * Executes the request with the provided resource and passed in variables.
-     *
-     * @param resource           The body of the script.
-     * @param parametersCallback The map of variables passed in from query that script in nested in.
-     * @throws ProviderException
-     */
-    @Override
-    public void executeScript(Resource resource, ParametersCallback parametersCallback) throws ProviderException {
-        run(resource, parametersCallback);
-
-        logger.debug("Operation complete for current entry.");
-    }
-
-    /**
-     * Executes the request as a query, returning the response. Not yet implemented.
-     *
-     * @param resource           The body of the query
-     * @param parametersCallback the map of variables.
-     * @param queryCallback      The response.
-     * @throws ProviderException
-     */
-    @Override
-    public void executeQuery(Resource resource, ParametersCallback parametersCallback, QueryCallback queryCallback) throws ProviderException {
-        throw new UnsupportedOperationException();
-    }
-
-    private void run(Resource resource, ParametersCallback parametersCallback) {
-
-        // Create request configurations and set timeout.
-        RequestConfig config = RequestConfig.custom()
-                .setConnectTimeout(timeOut)
-                .setConnectionRequestTimeout(timeOut)
-                .setSocketTimeout(timeOut).build();
-        httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
-        logger.debug("Built HttpClient");
-
-        ParametersCallbackMap parameters = new ParametersCallbackMap(parametersCallback);
-
-        try {
-            if (type.toUpperCase().equals("GET")) {
```

Comment:
So many duplicated constant, such as "GET", "PUT", "STRING".
Please provide either class or enum to standard them.